### PR TITLE
Drop py37 support

### LINKED ISSUE #12399: [🚀 Feature]: Drop `python3.7` support
### Feature and motivation

3.7 is now ~1 month passed EOL.  Dropping support will allow us to keep our dependencies better up to date and benefit from some of the new 3.8 goodies.

### Usage example

N/A

### PR DESCRIPTION
**Pull request by symonk**
Dropping support for `python3.7`.  This encompasses:

 * New python3.8+ syntax in places
 * Dropping 3.7 support in infrastructure code
 * Updating docs for any 3.7 references

closes #12399

### CODE REVIEW

**Comment by symonk** [comment]
not 100% the best approach to rolling this out, don't want to shock users using 3.7 without ample notice. But this is a start on the changes required

**Comment by symonk** [comment]
Todo: Update pypi classifiers for generated meta data (bazel)

**Comment by AutomatedTester** [comment]
Do we have a sense of how many folk are on 3.7? I know its either EOL or about to be EOL

**Comment by titusfortner** [comment]
Eol was end of June.
My big thing is bazel won't let us test with it, so we don't have an easy way to ensure what we have works with 3.7 anyway.

**Comment by AutomatedTester** [comment]
The syntax changes in 3.7->3.8 is partly to blame. I'm ok with dropping it fully since it's past EOL

**Review by AutomatedTester — APPROVED** [review]

**Comment by AutomatedTester** [comment]
In general this looks good. There are a number of conflicts

File: py/selenium/webdriver/common/print_page_options.py [hunk 1]
```diff
@@ -16,19 +16,14 @@
 # under the License.
 
 
-import sys
 from typing import TYPE_CHECKING
 from typing import List
 from typing import Optional
 
 # necessary to support types for Python 3.7
```
Comment (isaulv) [inline py/selenium/webdriver/common/print_page_options.py]: update comment

**Comment by isaulv** [comment]
Most of the changes are style related but it's good to use the more modern type annotations. This should be good to go once conflicts have been addressed.

**Comment by symonk** [comment]
Looping back to this now, however i've some conflicts to resolve still, will squash as a single commit when ironed out

File: py/selenium/webdriver/chromium/webdriver.py [hunk 2]
```diff
@@ -75,10 +75,9 @@ def launch_app(self, id):
     def get_network_conditions(self):
         """Gets Chromium network emulation settings.
 
-        :Returns:
-            A dict. For example:
-            {'latency': 4, 'download_throughput': 2, 'upload_throughput': 2,
-            'offline': False}
+        :Returns:     A dict. For example:     {'latency': 4,
```
Comment (AutomatedTester) [inline py/selenium/webdriver/chromium/webdriver.py:78]: Personally prefer the previous layout... was easier to read.

File: py/selenium/webdriver/common/timeouts.py [hunk 3]
```diff
@@ -39,9 +34,7 @@ class JSONTimeouts(TypedDict, total=False):
 class _TimeoutsDescriptor:
     """Get or set the value of the attributes listed below.
 
-    _implicit_wait
-    _page_load
-    _script
+    _implicit_wait _page_load _script
```
Comment (AutomatedTester) [inline py/selenium/webdriver/common/timeouts.py:37]: On a single line I think it becomes harder to read

**Review by AutomatedTester — APPROVED** [review]

File: py/selenium/webdriver/chromium/webdriver.py (hunk 2 above)
Comment (diemol) [reply py/selenium/webdriver/chromium/webdriver.py:78]: Tried to revert it but then the linter started to complain. I will leave it as it is.

File: py/selenium/webdriver/common/timeouts.py (hunk 3 above)
Comment (diemol) [reply py/selenium/webdriver/common/timeouts.py:37]: I tried to revert it but again the linter complained. I am not sure how it worked before.

**Review by diemol — APPROVED** [review]

**Pull request merged**